drivers/hmis/spec/models/hmis/hud/organization_access_spec.rb

```diff
@@ -61,5 +72,10 @@
       viewable_orgs = Hmis::Hud::Organization.viewable_by(user_with_p2_access)
       expect(viewable_orgs).to contain_exactly(o2)
     end
+
+    it 'includes only organizations associated with the data source ID of the current user' do
+      viewable_orgs = Hmis::Hud::Organization.viewable_by(user_with_ds1_and_ds2_access)
+      expect(viewable_orgs).to contain_exactly(o1, o2, o3)
```

Comment:
maybe just to drive the point home, we could add:
```ruby
# change hmis_data_source_id to ds2 to ensure it can access o4
user_with_ds1_and_ds2_access.hmis_data_source_id = ds2.id
viewable_orgs = Hmis::Hud::Organization.viewable_by(user_with_ds1_and_ds2_access)
expect(viewable_orgs).to contain_exactly(o4)

```